pennylane/clusters.py

```diff
@@ -0,0 +1,476 @@
+        measure (str, List[str]): Measurement(s) to perform. Options include
+            :func:`'expval' <pennylane.expval>`, :func:`'var' <pennylane.var>`,
+            and :func:`'sample' <pennylane.sample>`.
+            This can either be a single measurement type, in which case it is applied
+            to all observables, or a list of measurements of length ``len(observables)``.
+        interface (str, None): which interface to use for the returned QNode cluster.
+            This affects the types of objects that can be passed to/returned from the QNode:
+
+            * ``interface='autograd'``: Allows autograd to backpropogate
+              through the QNodeCluster. The QNodeCluster accepts default Python types
+              (floats, ints, lists) as well as NumPy array arguments,
+              and returns NumPy arrays.
+
+            * ``interface='torch'``: Allows PyTorch to backpropogate
+              through the QNodeCluster.The QNodeCluster accepts and returns Torch tensors.
+
+            * ``interface='tf'``: Allows TensorFlow in eager mode to backpropogate
+              through the QNodeCluster. The QNodeCluster accepts and returns
+              TensorFlow ``tf.Variable`` and ``tf.tensor`` objects.
+
+            * ``None``: The QNodeCluster accepts default Python types
+              (floats, ints, lists) as well as NumPy array arguments,
+              and returns NumPy arrays. It does not connect to any
+              machine learning library automatically for backpropagation.
+
+        diff_method (str, None): the method of differentiation to use in the created QNodeCluster.
+
+            * ``"best"``: Best available method. Uses the device directly to compute
+              the gradient if supported, otherwise will use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"parameter-shift"``: Use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"finite-diff"``: Uses numerical finite-differences.
+
+            * ``None``: a non-differentiable QNodeCluster is returned.
+
+    Returns:
+        QNodeCluster: a cluster of QNodes executing the circuit template with
+        the specified measurements
+
+    **Example:**
+
+    Let's define a custom two-wire template:
+
+    .. code-block:: python
+
+        def my_template(params, wires, **kwargs):
+            for i in range(2):
+                qml.RX(params[i], wires=wires[i])
+
+            qml.CNOT(wires=wires)
+
+    We want to compute the expectation values over the following list
+    of two-qubit observables:
+
+    >>> obs_list = [qml.PauliX(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliX(1)]
+
+    This can be easily done with the ``map`` function:
+
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> qnodes = qml.map(my_template, obs_list, dev, measure="expval")
```

Comment:
Perhaps specify the ``interface`` and ``diff_method`` here explicitly? Such that it is clear in the example as well, that those are something that can be defined (though might make the example too complicated).

Reply:
Yeah, I thought it would make the example too complicated (ideally, examples should be as minimal as possible to show working behaviour, and then users can refer to the arguments list for more advanced behaviour)